include/SFML/Window/ContextSettings.hpp

```diff
@@ -76,6 +81,8 @@ struct ContextSettings
     unsigned int majorVersion;      ///< Major number of the context version to create
     unsigned int minorVersion;      ///< Minor number of the context version to create
     Uint32       attributeFlags;    ///< The attribute flags to create the context with
+    std::string  vendorName;        ///< Name of the vendor of the OpenGL implementation
+    std::string  rendererName;      ///< Name of the OpenGL renderer implementation
```

Comment:
> I agree with Laurent, also the name "Context**Setting**" implies that it's something the user can set/configure.

> One alternative would be a new type like [`sf::Joystick::Identification`](https://github.com/LaurentGomila/SFML/blob/master/include/SFML/Window/Joystick.hpp#L76-L83), in addition to a `sf::Window::getDriverIdentification()` method or similar. Other ideas?

If the being a setting really implies being able to be set by the user then updating it with the "correct" data after a context is created doesn't make sense either. I could make the same argument and ask why not split all context information off into a separate data structure?

> What can a developer do with these extra information, other than output a warning, which we already do? In other words, is it really useful to expose it in the public API?

This is as useful/unuseful to expose as the context version, and we expose that already. There are many things a developer can do with this information, one of the most obvious ones being selecting a higher level of AA than 8, which translates to one of Nvidia's "special" AAs such as TXAA or FXAA. There are also other non-trivial differences between AMD and Nvidia OpenGL implementations that might hint the developer to optimize their rendering in different ways.

> To continue on what Bromeon said, I see this as an introspection fonction and as such should go along with functions to get the number of cpu and stuff like that. I feel it would be wiser to design this larger API first. I suggest therefore that we don't consider this feature for 2.3 in order not to stall its release.

Even if this is part of a wider API, where else would it be? In the system module? The introspection functions don't have to all be added at the same time, one can even consider the context version to be a limited form of introspection already.

The main reason I think this should be coupled with the 2.3 release is the fact that we have made it too easy to get away with a software context on Windows. The application would simply crash before, and at that point the developer might have asked for hardware/driver information. Now, it won't crash but simply exhibit unexpected behaviour which might not make the actual problem that obvious any more. If we don't provide them with a non-OpenGL way of getting this information they simply won't bother to check and instead come running to us with "bug reports" after 2.3 is released, kind of like what is already happening now to a limited degree.

Outright crashing when a software context was created before was just broken, which is why I fixed it. This is just the second half of the solution, and I'd prefer not splitting this into a separate release if possible.